apps/src/templates/projects/projectsRedux.js

```diff
@@ -302,6 +327,57 @@ export function updateProjectName(projectId, updatedName) {
   return {type: UPDATE_PROJECT_NAME, projectId, updatedName};
 }
 
+export function saveSuccess(projectId) {
+  return {type: SAVE_SUCCESS, projectId};
+}
+
+export function saveFailure(projectId) {
+  return {type: SAVE_FAILURE, projectId};
+}
+
+const fetchProjectToUpdate = (projectId, onComplete) => {
+  $.ajax({
+    url: `/v3/channels/${projectId}`,
+    method: 'GET',
+    type: 'json',
+    contentType: 'application/json;charset=UTF-8',
+  }).done((data) => {
+    onComplete(null, data);
+  }).fail((jqXhr, status) => {
+    onComplete(status, jqXhr.responseJSON);
+  });
+};
+
+const updateProjectNameOnServer = (project) => {
+  return (dispatch) => {
+    $.ajax({
+      url: `/v3/channels/${project.id}`,
+      method: 'POST',
+      type: 'json',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify(project)
+    }).done((data) => {
+      dispatch(saveSuccess(project.id));
+    }).fail((jqXhr, status) => {
+      dispatch(saveFailure(project.id));
```

Comment:
does this endpoint return a useful error in the `jqXhr` object? if it does, i think we would want to pass that error to `saveFailure` and set it in state so we can display a notification to the user about why updating the project name failed. this might be out of scope for what you're doing now, but good to consider

Reply:
`jqXhr.statusText` is useful-ish ("Bad Request").  I'll check with Poorva and see if we want a pop up message in this case or something else and do as a follow up to this PR.